fundingmanager.go

```diff
@@ -840,109 +838,125 @@ func (f *fundingManager) reservationCoordinator() {
 // advanceFundingState will advance the channel fromt the markedOpen state to
 // the point where the channel is ready for operation. This includes sending
 // funding locked to the peer, adding the channel to the router graph, and
-// announcing the channel.
-func (f *fundingManager) advanceFundingState(
-	channel *channeldb.OpenChannel) error {
-
-	channelState, shortChanID, err := f.getChannelOpeningState(
-		&channel.FundingOutpoint)
-	if err == ErrChannelNotFound {
-		// Channel not in fundingManager's opening database,
-		// meaning it was successfully announced to the
-		// network.
-		return nil
-	} else if err != nil {
-		return err
+// announcing the channel. The updateChan can be set non-nil to get
+// OpenStatusUpdates.
+//
+// NOTE: This MUST be run as a goroutine.
+func (f *fundingManager) advanceFundingState(channel *channeldb.OpenChannel,
+	updateChan chan<- *lnrpc.OpenStatusUpdate) {
+	defer f.wg.Done()
+
+	// We create the state-machine object which wraps the database state.
+	lnChannel, err := lnwallet.NewLightningChannel(
+		nil, channel, nil,
+	)
+	if err != nil {
+		fndgLog.Errorf("Unable to create LightningChannel(%v): %v",
+			channel.FundingOutpoint, err)
+		return
+	}
+
+	for {
+		channelState, shortChanID, err := f.getChannelOpeningState(
+			&channel.FundingOutpoint)
+		if err == ErrChannelNotFound {
+			// Channel not in fundingManager's opening database,
+			// meaning it was successfully announced to the
+			// network.
+			return
+		} else if err != nil {
+			fndgLog.Errorf("Unable to query database for "+
+				"channel opening state(%v): %v",
+				channel.FundingOutpoint, err)
+			return
+		}
+
+		// If we did find the channel in the opening state database, we
+		// have seen the funding transaction being confirmed, but there
+		// are still steps left of the setup procedure. We continue the
+		// procedure where we left off.
+		err = f.stateStep(channel, lnChannel, shortChanID,
```

Comment:
Folding should be: 
```go
err = f.stateStep(
    channel, lnChannel, shortChanID, channelState, 
    updateChan, 
)
```

Reply:
Fixed: https://github.com/lightningnetwork/lnd/pull/1910/commits/36ea8b14297150c4d35eeb60709e0e9c71b4c4a5